### PR TITLE
Fix deprecation in TraceableMetadataFactory

### PR DESCRIPTION
**Pull request by asuri0n**
https://github.com/schmittjoh/JMSSerializerBundle/issues/907

| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | no
| Doc updated   | no
| BC breaks?    | no
| Deprecations? | yes
| Fixed tickets | #907
| License       | MIT

### CODE REVIEW

File: Debug/TraceableMetadataFactory.php [hunk 1]
```diff
@@ -32,7 +34,7 @@ public function getAllClassNames(): array
         return $this->metadataFactory->getAllClassNames();
     }
 
-    public function getMetadataForClass(string $className)
+    public function getMetadataForClass(string $className): ClassHierarchyMetadata|MergeableClassMetadata|null
```
Comment (scyzoryck) [inline Debug/TraceableMetadataFactory.php:37]: It looks like the package still supports PHP 7.2 that do not support such syntax. May there other way to fix this deprecation message?

**Review by scyzoryck — CHANGES_REQUESTED** [review]
Thanks for contributing to this project! I left one comment on your changes.

File: Debug/TraceableMetadataFactory.php (hunk 1 above)
Comment (asuri0n) [reply Debug/TraceableMetadataFactory.php:37]: I'll check that

Comment (mbabker) [reply Debug/TraceableMetadataFactory.php:37]: Just like the deprecation message says, `add an explicit @return annotation to suppress this message`.  The B/C break can be dealt with later.

Comment (asuri0n) [reply Debug/TraceableMetadataFactory.php]: I've added @return annotation

**Review by scyzoryck — APPROVED** [review]

**Comment by ruudk** [comment]
Looks good! @goetas can this be merged and tagged? It prevents us from upgrading to 5.0

**Pull request merged**